src/main/scala/com/cognite/sdk/scala/common/OAuth2.scala

```diff
@@ -121,7 +121,11 @@ object OAuth2 {
   // scalastyle:on method.length
 
   object SessionProvider {
-    def apply[F[_]](session: Session, refreshSecondsBeforeTTL: Long = 30)(
+    def apply[F[_]](
+        session: Session,
+        refreshSecondsBeforeTTL: Long = 30,
+        tokenProvider: ServiceTokenProvider = new RealServiceTokenProvider()
```

Comment:
All we need is (love, but also) a function `getToken: F[String]` instead of using `ServiceTokenProvider` directly.

Maybe it should be `getServiceToken`, but what should an external customer pass to this? It's possible to use the CDF Session API externally, and this should (ideally) support that in a sensible way.